### PR TITLE
Use helper functions in a few more places

### PR DESCRIPTION
**Pull request by andy-ms**

### CODE REVIEW

File: src/services/completions.ts [hunk 1]
```diff
@@ -2202,18 +2202,8 @@ namespace ts.Completions {
...
-            if (!tags || position < pos || position > end) continue;
-            for (let i = tags.length - 1; i >= 0; i--) {
-                const tag = tags[i];
-                if (position >= tag.pos) {
-                    return tag;
-                }
-            }
-        }
+        return firstDefined((getJsDocHavingNode(node) as JSDocContainer).jsDoc, ({ pos, end, tags }) =>
+            pos <= position && position <= end ? findLast(tags, tag => position >= tag.pos) : undefined);
```
Comment (DanielRosenwasser) [inline src/services/completions.ts]: Pretty sure we have helper functions for determining inclusion of positions.

File: src/services/goToDefinition.ts [hunk 2]
```diff
@@ -257,13 +257,9 @@ namespace ts.GoToDefinition {
...
-    export function findReferenceInPosition(refs: ReadonlyArray<FileReference>, pos: number): FileReference {
-        for (const ref of refs) {
-            if (ref.pos <= pos && pos <= ref.end) {
-                return ref;
-            }
-        }
-        return undefined;
+
+    function findReferenceInPosition(refs: ReadonlyArray<FileReference>, pos: number): FileReference | undefined {
+        return find(refs, ref => ref.pos <= pos && pos <= ref.end);
```
Comment (DanielRosenwasser) [inline src/services/goToDefinition.ts]: Same here

File: src/services/completions.ts (hunk 1 above)
Comment (andy-ms) [reply src/services/completions.ts]: Added a `textRangeContainsPositionInclusive` helper.

**Comment by andy-ms** [comment]
@DanielRosenwasser Could you review again?

**Comment by andy-ms** [comment]
@DanielRosenwasser Could you review again?

**Pull request merged**